Neos.ContentRepository.TestSuite/Classes/Behavior/Features/Bootstrap/CRTestSuiteTrait.php

```diff
@@ -276,6 +263,31 @@ abstract protected function getContentRepositoryService(
         ContentRepositoryServiceFactoryInterface $factory
     ): ContentRepositoryServiceInterface;
 
+    final protected function getContentRepositoryServiceFactoryDependencies(): ContentRepositoryServiceFactoryDependencies
+    {
+        $accessorFactory = new class implements ContentRepositoryServiceFactoryInterface {
+            public function build(ContentRepositoryServiceFactoryDependencies $serviceFactoryDependencies): ContentRepositoryServiceInterface
+            {
+                return new class ($serviceFactoryDependencies) implements ContentRepositoryServiceInterface
+                {
+                    public function __construct(
+                        public readonly ContentRepositoryServiceFactoryDependencies $dependencies
+                    ) {
+                    }
+                };
+            }
+        };
+        return $this->getContentRepositoryService($accessorFactory)->dependencies;
+    }
+
+    final protected function getContentGraphReadModel(): ContentGraphReadModelInterface
```

Comment:
or maybe, if we like it or not, we should just expose the `ContentGraphReadModel` ON the content repository directly as `@internal` method with note to be careful. We cannot avoid people ditching security (see $context->withoutAuthorisationChecks) and as this `ContentGraphReadModel` will soon to be exposed for catchup hooks as api already ...

Reply:
IMO it's not about preventing folks to _intentionally_ circumvent security – they always can, as you pointed out.
But the API should make that cumbersome or at least very very explicit. And exposing a internal concept in an api-object is a code smell IMO (even if it is marked with warnings)

I think, we should just have some custom "accessor factory" in our `CRTestSuiteTrait` that is instantiated _once_  and provides read-access to all required dependencies.

Reply:
> that is instantiated _once_

Okay agreed, but that escalated to a bigger change: https://github.com/neos/neos-development-collection/pull/5341 so lets this optimisation separate for later.

Reply:
I would rather merge many small changes than always having these huge refactor everyhting things...

Reply:
oke so are you fine with the code now? @kitsunet